src/adaptors/sharpe-magnum/index.js

```diff
@@ -0,0 +1,88 @@
+const utils = require('../utils');
+const axios = require('axios');
+const ethers = require('ethers');
+const {abi} = require("./ABI")
+const {vaultAbi} = require("./vaultAbi")
+const sdk = require('@defillama/sdk');
+
+async function fetchSmaApr() {
+  const url = 'https://eth-api.lido.fi/v1/protocol/steth/apr/sma';
+  const response = await axios.get(url);
+  return response.data.data.smaApr;
+}
+
+async function getCurrentUserBorrowRate(vaultAddress) {
+
+  const poolToken = '0x030bA81f1c18d280636F32af80b9AAd02Cf0854e';
+
+  const rate = (
+    await sdk.api.abi.call({
+      target: "0x507fA343d0A90786d86C7cd885f5C49263A91FF4",
+      chain: "ethereum",
+      abi: abi.find((m) => m.name === 'getCurrentUserBorrowRatePerYear'),
+      params : [ poolToken, vaultAddress]
+    })
+  ).output;
+    
+  return ethers.BigNumber.from(rate);
+}
+
+async function apy(vaultAddress) {
+ 
+  const bRate = await getCurrentUserBorrowRate(vaultAddress);
+  //borrow rate is in ray (1E27)
+  let brate = bRate.div(ethers.BigNumber.from(10).pow(17));
+  let newBrate = parseFloat(brate.toString()) * 0.00000001;
+  const sRate = await fetchSmaApr();
+
+  const leverage = (
+    await sdk.api.abi.call({
+      target: vaultAddress,
+      chain: "ethereum",
+      abi: vaultAbi.find((m) => m.name === 'vaultsLeverage')
+    })
+  ).output;
+
+  // if vault working on 3x leverage
+  if(leverage == 2){
+    return (3.00 * sRate) - (2.00 * parseFloat(newBrate.toString()));
+  }
+  // if vault working on 2x leverage
+  else if(leverage == 1){
+    return (2.00 * sRate) - ( parseFloat(newBrate.toString()));
+  }
+  // if vault working on 1x leverage
+  else{
+    return sRate
+  } 
+  
+}
+
+async function getTVLInUSD() {
+  const url = 'https://api.llama.fi/tvl/sharpe-magnum';
```

Comment:
lets call the contracts here instead

Reply:
You mean instead of defillama api call, i should use my contract to fetch tvl??

Reply:
yes. i prefer to keep this separated especially if its just a simple call.

Reply:
Updated the script.